docs/ADR/2019-04-23-transcript-architecture.md

```diff
@@ -0,0 +1,170 @@
+# Architecture for Digital Paper Edit
+
+* Status: pending <!-- optional -->
+* Deciders: Pietro, Eimi <!-- Laurien, Mark Langton, James Gruessing -->
+* Date: 2019-04-23
+
+Technical Story: https://github.com/bbc/digital-paper-edit/issues/1 <!-- optional -->
+
+## Context and Problem Statement
+
+Digital Paper Edit has three milestones. Transcript, Paper-edit, and Annotation.
+We want to make sure that during the Transcript phase, we have a solid
+architecture to provide storage for uploaded content and transcription to be
+used in later phases. We want to consider technologies from a resilience,
+scalability, cost point of view, opensource compatibility and extensibility.
+
+## Decision Drivers
+
+* resilience
+* scalability
+* cost
+* number of users
+* opensource compatibility
+* extensibility
+
+## Considered Options
+
+* Framework and infrastructure: Docker vs Serverless vs Cosmos
+* Components: Lambda vs EC2 (AWS)
+* Storage: local vs S3
+* DB: PostgreSQL (RDS) vs DynamoDB vs MongoDB
+* Queues: STS vs RabbitMQ
+
+### Potential Problems in Options
+
+#### Framework and infrastructure: Serverless vs Docker vs Cosmos
+
+Serverless framework makes it easy to deploy serverless and
+many other components using a cli. It might be easy to begin
+with and is quick to deploy serverless, but becomes more difficult to maintain
+with transfer (local development vs production). It also isn't platform
+agnostic, which was what was initially suggested. It will give you the
+flexibility of using another platform, but you will still need to configure
+platform specific variables in your code. Developing and using this means we
+also lack transparency and resilience because the deployment occurs on someone's
+local development.
+
+Docker was considered to simplify the orchestration of microservices. Using docker means we could
+have a separation of concerns - the OS and the application. However, using this means we would still
+need to select a platform (AWS / GCP) and deploying it on the VM. This
+also implies that there is an extra level of abstraction: a Docker image (VM)
+inside a cloud VM. Debugging it also requires extra set up. If we decided to use
+Docker, we will not be able to use Cosmos which would give you the bulk of the
+configuration including HTTPS access for free.
+
+Using AWS and Cosmos means tying into the AWS architecture, makes it very
+difficult to move away. There is also overhead on learning how to use Cosmos.
+There is also less flexibility in the way we would do deployments - we would
+need to set up in Jenkins and be restricted to RHEL7, using RPMs as the
+packaging method.
+
+#### Components: Lambda vs EC2 (AWS)
+
+##### Lambda Pros and Cons
+
+From [this article](https://dzone.com/articles/the-pros-and-cons-of-aws-lambda),
+the pros and cons can be summarised as the following.
+
+Pros:
+
+* Reduced cost of execution
+* Improved application resiliency
+* Decouple server architecture to code
+* Idempotent
```

Comment:
```suggestion
* [Idempotent](https://en.wikipedia.org/wiki/Idempotence)
```

Did not come across this word before, though it was a misspelled version of independent. Hence the link suggestion.